config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_computetargettcpproxies.compute.cnrm.cloud.google.com.yaml

```diff
@@ -89,14 +89,19 @@ spec:
               description:
                 description: Immutable. An optional description of this resource.
                 type: string
+              location:
+                description: 'Location represents the geographical location of the
+                  ComputeTargetTCPProxy. Specify a region name or "global" for global
+                  resources. Reference: GCP definition of regions/zones (https://cloud.google.com/compute/docs/regions-zones/)'
+                type: string
               proxyBind:
                 description: |-
                   Immutable. This field only applies when the forwarding rule that references
                   this target proxy has a loadBalancingScheme set to INTERNAL_SELF_MANAGED.
                 type: boolean
               proxyHeader:
                 description: |-
-                  Specifies the type of proxy header to append before sending data to
+                  Immutable. Specifies the type of proxy header to append before sending data to
```

Comment:
This is generated by `make ready-pr`. The field is immutable in regional TCP proxy, but can be updated by `setProxyHeader` method in global TCP Proxy. 
@maqiuyujoyce currently, like other locational resources, we support regional and global in one ComputeTargetTCPProxy kind, is there a way to handle the case? Can we update the field description by ourselves anywhere? 
Or, maybe we should consider starting support ComputeTargetTCPProxy and RegionalComputeTargetTCPProxy kind separately?

Reply:
IMO it depends on the use case and the priority:
1. If it is a high priority to support this resource, and the common use cases include that the regional ComputeTargetTCPProxy referenced by other resources, then I suggest we do some hacks in the CRD generation to remove the `Immutable.` marking to unblock it asap.
2. If it is not super high priority (it's unclear from the ticket) or the resource won't be commonly referenced by other resources (it's referenced in ComputeForwardingRule), then we can do some comparison between the two resources of different locationalities, and decide as a team whether we want to support the regional one as a separate resource.

I actually don't recall much feedback about our locationality design, so it'll mainly be up to ourselves to make a decision based on how its pros and cons will impact direct controllers. We could reach out to the service team to gather some information though.

No matter which option we choose, I don't see it makes a big difference in direct controller migration later.